packages/util/src/signature.ts

```diff
@@ -22,12 +22,11 @@ export function ecsign(msgHash: Buffer, privateKey: Buffer, chainId?: bigint): E
   const r = Buffer.from(signature.slice(0, 32))
   const s = Buffer.from(signature.slice(32, 64))
 
-  if (chainId === undefined) {
-    const v = BigInt(recovery + 27)
-    return { r, s, v }
-  }
+  const v =
+    chainId === undefined
+      ? BigInt(recovery + 27)
+      : BigInt(recovery + 35) + BigInt(chainId) * BigInt(2)
```

Comment:
I'm not sure why this fixes it. These formulas look the same?

Reply:
This actually doesn't fix anything, I just tested to checkout the commit before and tests are still passing. I guess Gabriel meant the changes in the commit before to fix the situation.

Will re-read to try to give myself a confirmation on that.